arwen/contexts/runtime.go

```diff
@@ -833,6 +838,37 @@ func (context *runtimeContext) GetAllErrors() error {
 	return context.errors
 }
 
+// EnableGasTrace sets true to the flag variable that traces gas consumption
+func (context *runtimeContext) EnableGasTrace() {
+	context.traceGasEnabled = true
+}
+
+// DisableGasTrace sets true to the flag variable that traces gas consumption
+func (context *runtimeContext) DisableGasTrace() {
+	context.traceGasEnabled = false
+}
+
+// TraceGasUsed adds the usedGas passed to the traced gas map with the key as the passed functionName
+func (context *runtimeContext) TraceGasUsed(functionName string, initialGasLeft uint64) {
+	if context.traceGasEnabled {
+		if context.gasTrace[functionName] == nil {
+			context.gasTrace[functionName] = make([]uint64, 0)
+		}
+		usedGas := context.computeUsedGas(initialGasLeft)
```

Comment:
It would be better to compute the total Gas and give that one here, instead initialGasLeft

Reply:
I asked him for this method because it creates less copy-paste code in the hooks.
We could also have another method that just does the trace, without computing the delta.

Reply:
func (context *runtimeContext) TraceGas(functionName string, initialGasLeft uint64) { ... }

func (context *runtimeContext) TraceGasUsed(functionName string, initialGasLeft uint64) {
    context.TraceGas(context.computeUsedGas(initialGasLeft))
}

Reply:
This is not yet resolved.